libraries/AP_HAL_ChibiOS/hwdef/JHEMCU-GSF405A/readme.md

```diff
@@ -0,0 +1,95 @@
+# JHEMCU GSF405A AIO Flight Controller
+
+The JHEMCU GSF405A is a 1-2S Whoop-style (25.5mm x 25.5mm) flight controller featuring the STM32F405OG6 MCU, 4in1 5A ESCs, BMP280 barometer, 8MB flash, and an on-board ExpressLRS 2.4GHz receiver. The board weighs 3.3g, making it suitable for small or lightweight Ardupilot builds.
+
+![JHEMCU GSF405A](gsf405a.jpg "JHEMCU GSF405A Top")
+
+https://pyrodrone.com/collections/new-products/products/jhemcu-gsf405a-1s-2s-aio-f4-flight-control-w-5a-esc-elrs-2-4g-rx-25-5-25-5mm
+https://www.racedayquads.com/collections/new-products/products/jhemcu-gsf405a-1-2s-toothpick-whoop-aio-flight-controller-w-8bit-5a-esc-and-elrs-2-4ghz-rx
+
+### Specs:
+**Flight control parameters**
+* MCU: STM32F405OG6
+* Gyroscope/Accelerometer: MPU6000
+* OSD: AT7456E
+* Barometer: BMP280
+* Black box: 8MB
+* I2C: Support
+* BEC: 5V
+* UART: UART1 (ELRS), UART2 (external RC), UART3, UART4, UART6
+* USB: micro usb
+* Size: 25.5*25.5MM M2
+* Receiver: ELRS (CRSF), TBS (CRSF), SBUS, IBUS, DSM2, DSMX
+* Support programmable LED such as WS2812
+* Support buzzer
+* Built-in voltage and current sensors
+* Weight: 3.3 grams
+
+**ESC parameters**
+* Support PWM, Oneshot125, Oneshot42, Multishot, Dshot150, Dshot300, Dshot600
+* Input voltage: 1S-2S Lipo
+* Continuous current: 5A
+* Firmware: BLHELI_S S_H_50_REV16_7.HEX
+* *Note* Bidirectional DShot requires [flashing](https://esc-configurator.com/) a compatible ESC firmware (eg [Bluejay](https://github.com/mathiasvr/bluejay))). 
+
+## Pinout
+
+![GSF405A AIO Board](gsf405a_pinout.jpg "JHEMCU GSF405A Pinout")
+
+## UART Mapping
+
+The UARTs are marked Rn and Tn in the above pinouts. The Rn pin is the
+receive pin for UARTn. The Tn pin is the transmit pin for UARTn.
+|Name|Pin|Function|
+|:-|:-|:-|
+|SERIAL0|COMPUTER|USB|
+|SERIAL1|RX1/TX1|USART1 (DMA) - Used by the on-board ELRS receiver, but TX1/RX1 pads are also available |
+|SERIAL2|TX2/RX2|USART2 |
+|SERIAL3|TX3/RX3|USART3 (DMA) - GPS, located near I2C pads|
+|SERIAL4|TX4/RX4|UART4|
+|SERIAL6|TX6/RX6|USART6 (DMA) - Telemetry|
+
+## RC Input
+ 
+RC input is configured on SERIAL1 (USART1) connected to the ELRS receiver, configured for CRSF with `SERIAL1_PROTOCOL 23`. 
```

Comment:
you can set `SERIAL1_PROTOCOL 23` in the hwdef so it just works after flashing (i think)

Reply:
I moved it to  `defaults.parm` since the base hwdef.dat is used for both the ELRS and external RX targets. Should I move it back to the hwdef and just undefine it for the external RX target?

In what cases should `defaults.parm`be used versus defines in hwdef?